contracts/ColonyFunding.sol

```diff
@@ -95,6 +95,37 @@ contract ColonyFunding is ColonyStorage, PatriciaTreeProofs { // ignore-swc-123
     }
   }
 
+  function claimExpenditurePayout(uint256 _id, uint256 _slot, address _token) public
+  stoppable
+  expenditureExists(_id)
+  expenditureFinalized(_id)
+  {
+    Expenditure storage expenditure = expenditures[_id];
+    ExpenditureSlot storage slot = expenditureSlots[_id][_slot];
+    require(add(expenditure.finalizedTimestamp, slot.claimDelay) <= now, "colony-expenditure-cannot-claim");
+
+    FundingPot storage fundingPot = fundingPots[expenditure.fundingPotId];
+    assert(fundingPot.balance[_token] >= fundingPot.payouts[_token]);
+
+    uint256 payout = slot.payouts[_token];
+    uint256 payoutScalar = unpackPayoutScalar(slot.payoutScalar);
+    uint256 repPayout = wmul(payout, payoutScalar);
```

Comment:
Mm that is a good point, in my mind I had been considering the `MAX_PAYOUT` of `UINT254_MAX` as fine since the `MAX_PAYOUT_SCALAR` is 2, but it is in fact 2 WAD, which is different. Part of me would like to simply limit `MAX_PAYOUT` to `UINT128_MAX` and be done with it, since then a lot of these issues go away, and it's cleaner in my mind than `UINT254_MAX`.

Reply:
And out of curiousity, why do you dislike `wmul`? Is it because it gives perhaps a false sense of precision and inevitably results in some small rounding errors?

Reply:
> And out of curiosity, why do you dislike wmul?

From the documentation:

>// Wad arithmetic does not add orders of magnitude:
>
> wmul(1.1 ether, 2.2 ether) == 2.42 ether

The physicist in me says that should be 2.42 ether^2, and that playing so fast and loose with orders of magnitude/units is something that should be outlawed under international law 😛

Reply:
You're not wrong, but I do think that as long as `wmul` is used appropriately (i.e. one term is a unit-free scalar) then we're ok. Tools can be misused, but that doesn't make them bad tools.